### PR TITLE
feat: add support for helm chart

### PR DESCRIPTION
**Pull request by Revolyssup**
Screenshot 2024-10-06 at 11 49 20 PM
---
Screenshot 2024-10-06 at 11 49 05 PM

---
Screenshot 2024-10-06 at 11 39 34 PM

### CODE REVIEW

File: charts/values.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+controllerName: gateway.api7.io/api7-ingress-controller
+replicas: 1
+admin:
+  key: '' # Pass the admin key generated for the ingress gateway group
+  adminTLSVerify: false
+  adminEndpoint: https://api7ee3-dashboard.api7-ee-e2e:7443/apisix/admin
+namespace: ingress-apisix
```
Comment (AlinsRan) [inline charts/values.yaml:7]: Please use the `.Release.Namespace` and do not specify namespace in the value

File: charts/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ingress-config
```
Comment (AlinsRan) [inline charts/templates/configmap.yaml:4]: Using a fixed name may be mistakenly deleted by other releases.

please add:
```yaml
  name: {{ .Release.Name }}-xxx
  namespace: {{ .Release.Namespace }} 
 ```

File: charts/templates/cluster_role_binding.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app.kubernetes.io/name: api7-ingress
+  name: api7-ingress-manager-rolebinding
```
Comment (AlinsRan) [inline charts/templates/cluster_role_binding.yaml:6]: please add prefix:
`{{ .Release.Name }}`

File: charts/templates/cluster_role.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: api7-ingress-manager-role
```
Comment (AlinsRan) [inline charts/templates/cluster_role.yaml:4]: please add prefix:
{{ .Release.Name }}

File: charts/templates/cluster_role.yaml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  - httproutes/status
+  verbs:
+  - get
+  - update
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: api7-ingress-metrics-auth-role
```
Comment (AlinsRan) [inline charts/templates/cluster_role.yaml:109]: please add prefix:
{{ .Release.Name }}

File: charts/templates/cluster_role.yaml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: api7-ingress-metrics-reader
```
Comment (AlinsRan) [inline charts/templates/cluster_role.yaml:127]: please add prefix:
{{ .Release.Name }}

File: charts/values.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+statusAddress: ''
```
Comment (AlinsRan) [inline charts/values.yaml:8]: Need log level.

File: charts/values.yaml (hunk 1 above)
Comment (AlinsRan) [reply charts/values.yaml:7]: remove it.

File: charts/values.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+controllerName: gateway.api7.io/api7-ingress-controller
+replicas: 1
+admin:
+  key: '' # Pass the admin key generated for the ingress gateway group
+  adminTLSVerify: false
+  adminEndpoint: https://api7ee3-dashboard.api7-ee-e2e:7443/apisix/admin
```
Comment (AlinsRan) [inline charts/values.yaml:6]: It is at the admin level, so admin should be removed.

File: charts/values.yaml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+controllerName: gateway.api7.io/api7-ingress-controller
+replicas: 1
+admin:
+  key: '' # Pass the admin key generated for the ingress gateway group
```
Comment (AlinsRan) [inline charts/values.yaml:4]: Suggest using `required` to constrain it

File: charts/templates/service.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/name: api7-ingress
+    control-plane: controller-manager
+  name: {{ .Release.Name }}-metrics-service
+  namespace: {{ .Values.namespace }}
```
Comment (AlinsRan) [inline charts/templates/service.yaml:8]: `{{ .Release.Namespace }}`

File: charts/templates/service.yaml [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  ports:
+  - name: https
+    port: 8443
+    protocol: TCP
+    targetPort: 8443
+  selector:
```
Comment (AlinsRan) [inline charts/templates/service.yaml:15]: please ref: https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix-ingress-controller/templates/_helpers.tpl

Labels and selector cannot be a fixed value.

File: charts/values.yaml (hunk 7 above)
Comment (AlinsRan) [reply charts/values.yaml:8]: Need image tag.

**Review by AlinsRan — CHANGES_REQUESTED** [review]

**Comment by AlinsRan** [comment]
I suggest you take a closer look at the implementation of the open-source helm chart.

https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix-ingress-controller

File: charts/templates/service.yaml (hunk 11 above)
Comment (Revolyssup) [reply charts/templates/service.yaml:14]: done

**Comment by Revolyssup** [comment]
> I suggest you take a closer look at the implementation of the open-source helm chart.
> 
> https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix-ingress-controller
> 
> There is a lot of content inside that can be used as a reference for us.

Do we need all the configuration options in values.yaml as in this helm chart? Or few from here?

**Comment by AlinsRan** [comment]
> > I suggest you take a closer look at the implementation of the open-source helm chart.
> > https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix-ingress-controller
> > There is a lot of content inside that can be used as a reference for us.
> 
> Do we need all the configuration options in values.yaml as in this helm chart? Or few from here?

Only a few key ones are needed, and the rest will be added as needed later.
You need to test if it is possible to install multiple ingress in the same namespace.

Then uninstall one of the controllers and confirm if it will affect the operation of the other controllers.

**Comment by Revolyssup** [comment]
@AlinsRan I have added the important configurables. If I missed something you can point out. I have also checked with this.
> You need to test if it is possible to install multiple ingress in the same namespace.

> Then uninstall one of the controllers and confirm if it will affect the operation of the other controllers.

File: charts/values.yaml [hunk 12]
```diff
@@ -0,0 +1,37 @@
+nameOverride: ""
+labelsOverride: {}
+annotations: {}
+podAnnotations: {}
+controllerName: gateway.api7.io/api7-ingress-controller
+replicas: 1
+admin:
+  key: 'a7adm-e8J8XAu6FX3W9LD1f84dg3JCpTIj22e733p5qLZ8NY304ksZIz-1ff6f82c482a41f6b8d86af79d3a1c20' # Pass the admin key generated for the ingress gateway group
+  tlsVerify: false
+  endpoint: https://api7ee3-dashboard.api7-ee-e2e:7443/apisix/admin
```
Comment (AlinsRan) [inline charts/values.yaml:10]: Can't the key and endpoint be removed? They don't need default values.

File: charts/values.yaml [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+statusAddress: ''
+logLevel: debug
```
Comment (AlinsRan) [inline charts/values.yaml:12]: info

File: charts/templates/cluster_role_binding.yaml [hunk 14]
```diff
@@ -0,0 +1,27 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app.kubernetes.io/name: api7-ingress
+  name: {{ .Release.Name }}-{{ .Release.Name }}-api7-ingress-manager-rolebinding
```
[the remark this answers is not in the document]
Comment (Revolyssup) [reply charts/templates/cluster_role_binding.yaml:6]: fixed

File: charts/values.yaml (hunk 13 above)
Comment (Revolyssup) [reply charts/values.yaml:12]: done

File: charts/values.yaml (hunk 12 above)
Comment (Revolyssup) [reply charts/values.yaml:10]: done

File: charts/templates/cluster_role_binding.yaml (hunk 14 above)
[the remark this answers is not in the document]
Comment (Revolyssup) [reply charts/templates/cluster_role_binding.yaml:6]: done

**Comment by AlinsRan** [comment]
1. Need to add RBAC to the crd.
2. It is recommended to change the file name to: `gateway.aisix.io_plugingconfig.yaml`

**Comment by Revolyssup** [comment]
> 1. RBAC permissions need to be set for the CRDs.
> 2. It is recommended to change the file name to: `gateway.aisix.io_plugingconfig.yaml`

done

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**